src/Agent.Listener/DistributedTask.Pipelines/Yaml/Contracts/CheckoutStep.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.TypeConverters;
+
+namespace Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.Contracts
+{
+    internal sealed class CheckoutStep : ISimpleStep
+    {
+        public String Name { get; set; }
+
+        internal String Clean { get; set; }
```

Comment:
Cool. Fyi - here is the [entry point for the yaml getting started docs](https://github.com/Microsoft/vsts-agent/blob/master/docs/preview/yamlgettingstarted.md). It's not available yet, but this is where we are heading. I need to update the docs... this PR is changing the schema to simplify some things. I am switching focus right now to update the docs as part of this PR because the schema is changing a little.